ccm_web/client/src/pages/AddNonUMUsers.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState } from 'react'
+import { Backdrop, CircularProgress, Grid, makeStyles, Typography } from '@material-ui/core'
+
+import * as api from '../api'
+import ErrorAlert from '../components/ErrorAlert'
+import Help from '../components/Help'
+import MultipleUserEnrollmentWorkflow from '../components/MultipleUserEnrollmentWorkflow'
+import UserEnrollmentForm from '../components/UserEnrollmentForm'
+import MethodSelect from '../components/MethodSelect'
+import usePromise from '../hooks/usePromise'
+import {
+  CanvasCourseSection, CanvasCourseSectionWithCourseName, getRolesUserCanEnroll, injectCourseName,
+  sortSections
+} from '../models/canvas'
+import { CCMComponentProps } from '../models/FeatureUIData'
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    padding: 25,
+    textAlign: 'left'
+  },
+  spacing: {
+    marginBottom: theme.spacing(2)
+  },
+  container: {
+    position: 'relative',
+    zIndex: 0
+  },
+  backdrop: {
+    zIndex: theme.zIndex.drawer + 1,
+    color: '#FFF',
+    position: 'absolute',
+    textAlign: 'center'
+  }
+}))
+
+type InputMethod = 'single' | 'csv'
+
+enum PageState {
+  SelectInputMethod,
+  AddSingleUser,
+  AddCSVUsers
+}
+
+interface AddNonUMUsersProps extends CCMComponentProps {}
+
+export default function AddNonUMUsers (props: AddNonUMUsersProps): JSX.Element {
+  const classes = useStyles()
+
+  const { course, canvasURL } = props.globals
+  if (course.roles.length === 0) return <ErrorAlert />
+
+  const settingsURL = `${canvasURL}/courses/${course.id}/settings`
+  const rolesUserCanEnroll = getRolesUserCanEnroll(course.roles)
+
+  const [activePageState, setActivePageState] = useState<PageState>(PageState.SelectInputMethod)
+  const [inputMethod, setInputMethod] = useState<InputMethod>('single')
+  const [sections, setSections] = useState<CanvasCourseSectionWithCourseName[] | undefined>(undefined)
+
+  const [doGetSections, isGetSectionsLoading, getSectionsError, clearGetSectionsError] = usePromise(
+    async () => await api.getCourseSections(props.globals.course.id),
+    (sections: CanvasCourseSection[]) => setSections(injectCourseName(sections, props.course.name))
+  )
+
+  useEffect(() => {
+    if (sections === undefined && getSectionsError === undefined) {
+      void doGetSections()
+    }
+  }, [sections, getSectionsError])
+
+  const getSectionsErrorAlert = (
+    <ErrorAlert
+      messages={
+        [<Typography key={0}>An error occurred while loading section data from Canvas.</Typography>]
+      }
+      tryAgain={clearGetSectionsError}
+    />
+  )
```

Comment:
For Future/Icebox: I feel that the L60-L78, can be extracted to as separate component and can be reused in features like where ever the sections list is needed.

Reply:
See issue #296.